be/src/exec/es_http_scan_node.cpp

```diff
@@ -0,0 +1,442 @@
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "exec/es_http_scan_node.h"
+
+#include <chrono>
+#include <sstream>
+
+#include "common/object_pool.h"
+#include "exprs/expr.h"
+#include "runtime/runtime_state.h"
+#include "runtime/row_batch.h"
+#include "runtime/dpp_sink_internal.h"
+#include "service/backend_options.h"
+#include "util/runtime_profile.h"
+#include "util/es_scan_reader.h"
+#include "util/es_scroll_query.h"
+#include "util/es_query_builder.h"
+#include "exec/es_predicate.h"
+
+namespace doris {
+
+EsHttpScanNode::EsHttpScanNode(
+        ObjectPool* pool, const TPlanNode& tnode, const DescriptorTbl& descs) : 
+            ScanNode(pool, tnode, descs), 
+            _tuple_id(tnode.es_scan_node.tuple_id),
+            _runtime_state(nullptr),
+            _tuple_desc(nullptr),
+            _num_running_scanners(0),
+            _scan_finished(false),
+            _eos(false),
+            _max_buffered_batches(1024),
+            _wait_scanner_timer(nullptr) {
+}
+
+EsHttpScanNode::~EsHttpScanNode() {
+}
+
+Status EsHttpScanNode::init(const TPlanNode& tnode, RuntimeState* state) {
+    RETURN_IF_ERROR(ScanNode::init(tnode));
+
+    // use TEsScanNode
+    _properties = tnode.es_scan_node.properties;
+    return Status::OK;
+}
+
+Status EsHttpScanNode::prepare(RuntimeState* state) {
+    VLOG_QUERY << "EsHttpScanNode prepare";
+    RETURN_IF_ERROR(ScanNode::prepare(state));
+
+    _runtime_state = state;
+    _tuple_desc = state->desc_tbl().get_tuple_descriptor(_tuple_id);
+    if (_tuple_desc == nullptr) {
+        std::stringstream ss;
+        ss << "Failed to get tuple descriptor, _tuple_id=" << _tuple_id;
+        return Status(ss.str());
+    }
+
+    // set up column name vector for ESScrollQueryBuilder
+    for (auto slot_desc : _tuple_desc->slots()) {
+        if (!slot_desc->is_materialized()) {
+            continue;
+        }
+        _column_names.push_back(slot_desc->col_name());
+    }
+
+    _wait_scanner_timer = ADD_TIMER(runtime_profile(), "WaitScannerTime");
+
+    return Status::OK;
+}
+
+// build predicate 
+Status EsHttpScanNode::build_conjuncts_list() {
+    Status status = Status::OK;
+    for (int i = 0; i < _conjunct_ctxs.size(); ++i) {
+        EsPredicate* predicate = _pool->add(
+                    new EsPredicate(_conjunct_ctxs[i], _tuple_desc));
+        if (predicate->build_disjuncts_list()) {
+            _predicates.push_back(predicate);
+            _predicate_to_conjunct.push_back(i);
+        } else if (!predicate->get_es_query_status().ok()) {
+            return predicate->get_es_query_status();
```

Comment:
need some LOG to see why this is not `ok()`

Reply:
Done